### DISCUSSION TITLE
[Feature] Mobile app allow upload from WhatsApp

### DISCUSSION THREAD

**Discussion by mkjustuk** [immich-app/immich#2843] [Feature Request]
### The feature

Often, I'm sent images I'd like to add to my library (another family member took a photo while I took a video at a birthday party etc).

It would be awesome if the share external option in WhatsApp could interface with the Immich app to upload that image/video into the library there and then.

Alternatively, if there was an option in the Immich app to adhoc upload from the WhatsApp image storage.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by ThisIsTheOnlyUsernameAvailable** [comment]
WhatsApp saves photos you're sent in a folder on your phone (well, it does on Android anyway).  So you just include that folder in those that you sync to Immich, and it's all automatic.

**Reply by mkjustuk** [reply]
Thanks, yes I saw that when I added my camera folder.  But I don´t want to do that, as I get sent all kinds of garbage on WhatsApp that I would never want in my library on immich so need to be (very) selective.

**Reply by bo0tzz** [reply]
If you select 'Save' on an image in WhatsApp, it gets saved to a separate location on your phone's storage which you can then add to Immich.

**Reply by mkjustuk** [reply]
Unfortunately that doesn´t work as the dates get all screwed up if you don´t do it the same day.

**Comment by bo0tzz** [comment]
Duplicate of #2440
